src/test/java/io/cdap/plugin/sfmc/sink/RecordDataExtensionRowConverterTest.java

```diff
@@ -110,5 +110,105 @@ public void testTransform() throws ETSdkException {
+    ETDataExtensionColumn column = new ETDataExtensionColumn();
+    column.setId("121");
+    column.setDescription("newCustomer");
+    column.setLength(22);
+    Date d = new Date(2022, 02, 9);
+    column.setCreatedDate(d);
+    column.setDefaultValue("default");
+    column.setIsPrimaryKey(true);
+    column.setIsRequired(true);
+    column.setKey("key");
+    column.setModifiedDate(d);
+    column.setName("name");
+    column.setModified("date", false);
+    column.setPrecision(0);
+    column.setScale(100);
+    columnList.add(column);
+    Map<String, ETDataExtensionColumn> columns = new HashMap<>();
+    columns.put("tableName", column);
+    String externalKey = "externalKey";
+    DataExtensionInfo dataExtensionInfo = new DataExtensionInfo(externalKey, columnList);
+    RecordDataExtensionRowConverter recorddataextensionrowconverter =
+      new RecordDataExtensionRowConverter(dataExtensionInfo, true);
+    try {
+      Assert.assertEquals(recorddataextensionrowconverter.getType(type, fieldName, val1), strVal);
+    } catch (Exception e) {
+    }
+  }
+  @Test
+  public void testTransformDefault() {
+    List<ETDataExtensionColumn> columnList = new ArrayList<>();
+    ETDataExtensionColumn column1 = new ETDataExtensionColumn();
+    column1.setName("storeid");
+    column1.setName("emailid");
+    column1.setName("timestamp");
+    ETDataExtensionColumn column2 = new ETDataExtensionColumn();
+    column2.setName("abcd");
+    columnList.add(column1);
+    columnList.add(column2);
+    Schema inputSchema = Schema.recordOf("record",
+                                         Schema.Field.of("storeid", Schema.of(Schema.Type.STRING)),
+                                         Schema.Field.of("emailid", Schema.of(Schema.Type.STRING)),
+                                         Schema.Field.of("timestamp",
+                                                         Schema.nullableOf(Schema.of
+                                                           (Schema.LogicalType.TIMESTAMP_MICROS))),
+                                         Schema.Field.of("abcd", Schema.of(Schema.LogicalType.DATETIME)));
+    DataExtensionInfo dataExtensionInfo = new DataExtensionInfo("DE", columnList);
+    RecordDataExtensionRowConverter recordDataExtensionRowConverter =
+      new RecordDataExtensionRowConverter(dataExtensionInfo, false);
+    StructuredRecord structuredRecord = Mockito.mock(StructuredRecord.class);
+    Mockito.when(structuredRecord.getSchema()).thenReturn(inputSchema);
+    Mockito.when(structuredRecord.get("emailid")).thenReturn("abc@123");
+    LocalDate date = LocalDate.of(2011, 01, 01);
+    Mockito.when(structuredRecord.get("abcd")).thenReturn(date);
+    Mockito.when(structuredRecord.getDate("abcd")).thenReturn(date);
+    Mockito.when(structuredRecord.getDecimal("price")).thenReturn(BigDecimal.valueOf(2000));
+    List<String> list = new ArrayList<>();
+    list.add("abcd");
+    try {
```

Comment:
Ideally Assert statements should not be put inside the try-catch block. Please make sure to move all the asserts outside the try-catch. In case we do expect an exception to be thrown make that as a new test case which repros the exception scenario and expects and exception in the process.

Reply:
Yes, In those test cases we do expect an exception to be thrown. However, I have done the changes as per the need with exception approach. Added comments as well. Please have a look.